harness/determined/tensorboard/base.py

```diff
@@ -95,3 +97,30 @@ def get_metric_writer() -> tensorboard.BatchMetricWriter:
         writer = pytorch.TorchWriter()
 
     return tensorboard.BatchMetricWriter(writer)
+
+
+class _TensorboardUploadThread(threading.Thread):
+    def __init__(
+        self, upload_function: Callable[[List[pathlib.Path]], None], work_queue: queue.Queue
+    ) -> None:
+        self._upload_function = upload_function
+        self._work_queue = work_queue
+
+        super().__init__()
+
+    def run(self) -> None:
+        while True:
+            # The thread will wait until self._work_queue is not empty
+            file_paths = self._work_queue.get(block=True, timeout=None)
```

Comment:
No need to pass any parameters here, these are the default behaviors.

Reply:
Will do so!